synctl

```diff
@@ -3995,19 +4153,36 @@ def main():
-                updated_payload = update_instance.get_json()
+                updated_payload = update_instance.get_updated_test_config()
                 update_instance.update_a_synthetic_test(get_args.id, updated_payload)
         if get_args.syn_type == SYN_ALERT:
-            invalid_options = [get_args.active, get_args.frequency, get_args.timeout, get_args.retry_interval, get_args.retries,
-                               get_args.operation, get_args.script_file, get_args.location, get_args.record_video,
-                               get_args.mark_synthetic_call, get_args.entry_file, get_args.url, get_args.follow_redirect,
-                               get_args.expect_status, get_args.validation_string, get_args.bundle, get_args.custom_property]
-            for option in invalid_options:
-                if option is not None:
-                    print("not supported by smart alert")
+            invalid_options = ["label", "active", "frequency", "timeout", "retry_interval", "retries", "operation", "script_file",
+                               "location", "record_video", "mark_synthetic_call", "entry_file", "url", "follow_redirect",
+                               "expect_status", "validation_string", "bundle", "custom_property"]
+            update_instance.invalid_update_options(invalid_options, update_args)
+            payload = alert_instance.retrieve_a_smart_alert(get_args.id)
+            update_alert.set_updated_payload(payload)
             if get_args.file is not None:
-                new_payload = update_instance.update_using_file(get_args.file)
-                update_instance.update_a_smart_alert(get_args.file, new_payload)
+                new_payload = update_alert.update_using_file(get_args.file)
+                update_alert.update_a_smart_alert(get_args.id, new_payload)
+            else:
+                if get_args.name is not None:
+                    update_alert.update_alert_name(get_args.name)
+                if get_args.description is not None:
+                    update_alert.update_description(get_args.description)
+                if get_args.severity is not None:
+                    update_alert.update_alert_severity(get_args.severity)
+                if get_args.alert_channel is not None:
+                    update_alert.update_alert_channel(get_args.alert_channel)
+                if get_args.test is not None:
+                    update_alert.update_tests(get_args.test)
+                if get_args.violation_count is not None:
+                    update_alert.update_violation_count(get_args.violation_count)
+                updated_alert_config = update_alert.get_updated_alert_config()
+                update_alert.update_a_smart_alert(get_args.id, updated_alert_config)
+                operation = "enable" if get_args.enable else "disable" if get_args.disable else None
+                if operation is not None:
+                    update_alert.toggle_smart_alert(get_args.id, operation)
```

Comment:
`update_a_smart_alert` and `toggle_smart_alert` seems duplicated here, can we use `update_a_smart_alert` to toggle an alert ?
```
$ synctl update alert 2ByoISo6TPqAy4VFdZrRDg --enable -e pink
alert 2ByoISo6TPqAy4VFdZrRDg did not change
alert 2ByoISo6TPqAy4VFdZrRDg enabled

$ synctl update alert 2ByoISo6TPqAy4VFdZrRDg --disable -e pink
alert 2ByoISo6TPqAy4VFdZrRDg did not change
alert 2ByoISo6TPqAy4VFdZrRDg disabled
```

Reply:
`update_a_smart_alert`  uses POST to update, it doesn't support enable/disable. To toggle alert we use PUT

Reply:
I see, but logs looks confusable, we can move disable/enable to patch command when we add patch for Smart Alert.

Reply:
but I don't think alerts supports patch as well, can we separate `enable/disable` from other update commands?

Reply:
I think so, separate it from other update options, if `enable/disable` is specified, call `toggle_smart_alert` , else call `update_a_smart_alert`